### PR TITLE
correct summing total blocks in EfficientNet

### PR DESCRIPTION
**Pull request by yixingfu**
The number of total blocks incorrect, as it is not updated for the depth_coefficient. As an effect, it will  make drop_rate much higher than expected. In the case of larger base models (for example B3), choosing drop_connect_rate to be large (for example 0.8) will result in dropout rate for some layers becoming larger than 1.

### CODE REVIEW

**Comment by qlzh727** [comment]
Thanks for reporting the issue and sending the PR.

I found the link of the original implementation from the https://arxiv.org/abs/1905.11946, and corresponding code is at https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/efficientnet_model.py#L693.

```
drop_rate = 1.0 - survival_prob
survival_prob = 1.0 - drop_rate * float(idx) / len(self._blocks)
```

self._blocks is the list that contains all the blocks, the block is created by rounded value of "repeat".  The existing implementation is incorrect since adding unrounded value together could be larger than the total number of blocks.

**Comment by qlzh727** [comment]
Since this change will cause potential change to the saved weights. Could you check if the existing h5 weights need any update with the default parameter?

**Comment by yixingfu** [comment]
I think if the weights are pre-trained using this implementation, then it has to be updated except for B0 where depth scaling is 1.  

The difference here is equivalent to changing `drop_connect_rate` value. Varying drop_connect_rate does have effect on training result (although it would probably not visibly alter the performance of the model). However, using the given pre-trained weights, varying drop_connect_rate gives exactly the same model. That means the change in drop_connect_rate cannot be accounted for without updating the weight files.

I tried to make a more explicit comparison between the existing and corrected code, but building TF from source is too much pain - is there any better way to test?

**Comment by qlzh727** [comment]
Yea, I think that is aligning with my expectation as well, we will have to update the weights. We need to check with @fchollet about how weights are created and uploaded to Keras side. There is also a checksum in the source code, which I think need to be updated within the PR as well.

**Comment by qlzh727** [comment]
@yixingfu discovered that the keras-application weights are exactly same as the one from the checkpoint in https://github.com/tensorflow/tpu/tree/30b0889ee1933c3dbe90f1c46461a1a89370a5ba/models/official/efficientnet, which means the weights doesn't need to be updated in this PR. Approving this PR.

**Review by qlzh727 — DISMISSED** [review]

**Review by qlzh727 — APPROVED** [review]
